packages/go_router/lib/src/state.dart

```diff
@@ -69,10 +72,63 @@ class GoRouterState {
   /// A unique string key for this sub-route, e.g. ValueKey('/family/:fid')
   final ValueKey<String> pageKey;
 
+  /// Gets the [GoRouterState] from context.
+  ///
+  /// The returned [GoRouterState] will depends on which [GoRoute] or
+  /// [ShellRoute] the input `context` is in.
+  ///
+  /// This method only supports [GoRoute] and [ShellRoute] that generate
+  /// [ModalRoute]s. This is typically the case if one uses [GoRoute.builder],
+  /// [ShellRoute.builder], [CupertinoPage], [MaterialPage],
+  /// [CustomTransitionPage], or [NoTransitionPage].
+  ///
+  /// This method is fine to be called during [GoRoute.builder] or
+  /// [ShellRoute.builder].
+  ///
+  /// This method cannot be called during [GoRoute.pageBuilder] or
+  /// [ShellRoute.pageBuilder] since there is no [GoRouterState] to be
+  /// associated with.
+  ///
+  /// To access GoRouterState from a widget.
+  ///
+  /// ```
+  /// GoRoute(
+  ///   path: '/:id'
+  ///   builder: (_, __) => MyWidget(),
+  /// );
+  ///
+  /// class MyWidget extends StatelessWidget {
+  ///   @override
+  ///   Widget build(BuildContext context) {
+  ///     return Text('${GoRouterState.of(context).params['id']}');
+  ///   }
+  /// }
+  /// ```
+  static GoRouterState of(BuildContext context) {
+    final ModalRoute<Object?>? route = ModalRoute.of(context);
+    if (route == null) {
+      throw GoError('There is no modal route above the current context.');
+    }
+    final RouteSettings settings = route.settings;
+    if (settings is! Page<Object?>) {
+      throw GoError(
+          'The parent route must be a page route to have a GoRouterState');
+    }
+    final GoRouterStateRegistryScope? scope = context
+        .dependOnInheritedWidgetOfExactType<GoRouterStateRegistryScope>();
+    if (scope == null) {
+      throw GoError(
+          'There is no GoRouterStateRegistryScope above the current context.');
+    }
+    final GoRouterState state =
+        scope.notifier!._createPageRouteAssociation(settings, route);
+    return state;
+  }
+
   /// Get a location from route name and parameters.
   /// This is useful for redirecting to a named location.
-  // TODO(johnpryan): deprecate namedLocation API
-  // See https://github.com/flutter/flutter/issues/10772
+  @Deprecated(
+      'Uses GoRouter.of(context).routeInformationParser.namedLocation instead')
```

Comment:
@chunhtai 

`GoRouteInformationParser` doesn't have `namedLocation`.
https://github.com/flutter/packages/blob/main/packages/go_router/lib/src/parser.dart

What is the correct alternative?